packages/core/helper-plugin/src/features/AutoReloadOverlayBlocker.js

```diff
@@ -0,0 +1,248 @@
+    }
+  }, [isOpen]);
+
+  let displayedIcon = config?.icon || 'reload';
+
+  let description = {
+    id: config?.description || 'components.OverlayBlocker.description',
+    defaultMessage:
+      "You're using a feature that needs the server to restart. Please wait until the server is up.",
+  };
+
+  let title = {
+    id: config?.title || 'components.OverlayBlocker.title',
+    defaultMessage: 'Waiting for restart',
+  };
+
+  if (failed) {
+    displayedIcon = 'time';
+
+    description = {
+      id: 'components.OverlayBlocker.description.serverError',
+      defaultMessage: 'The server should have restarted, please check your logs in the terminal.',
+    };
+
+    title = {
+      id: 'components.OverlayBlocker.title.serverError',
+      defaultMessage: 'The restart is taking longer than expected',
+    };
+  }
+
+  const autoReloadValue = React.useMemo(
+    () => ({
+      lockApp: lockAppWithAutoreload,
+      unlockApp: unlockAppWithAutoreload,
+    }),
+    [lockAppWithAutoreload, unlockAppWithAutoreload]
+  );
+
+  return (
+    <AutoReloadOverlayBlockerContext.Provider value={autoReloadValue}>
+      <Blocker
+        displayedIcon={displayedIcon}
+        isOpen={isOpen}
+        description={description}
+        title={title}
+      />
+      {children}
+    </AutoReloadOverlayBlockerContext.Provider>
+  );
+};
+
+AutoReloadOverlayBlockerProvider.propTypes = {
+  children: PropTypes.element.isRequired,
+};
+
+const Blocker = ({ displayedIcon = 'reload', description, title, isOpen }) => {
+  const { formatMessage } = useIntl();
+
+  // eslint-disable-next-line no-undef
+  return isOpen && globalThis?.document?.body
+    ? createPortal(
+        <Overlay id="autoReloadOverlayBlocker" direction="column" alignItems="center" gap={6}>
+          <Flex direction="column" alignItems="center" gap={2}>
+            <Typography as="h1" variant="alpha">
+              {formatMessage(title)}
+            </Typography>
+            <Typography as="h2" textColor="neutral600" fontSize={4} fontWeight="regular">
+              {formatMessage(description)}
+            </Typography>
+          </Flex>
+          {displayedIcon === 'reload' && (
+            <IconBox padding={6} background="primary100" borderColor="primary200">
+              <LoaderReload width={pxToRem(36)} height={pxToRem(36)} />
+            </IconBox>
+          )}
+          {displayedIcon === 'time' && (
+            <IconBox padding={6} background="primary100" borderColor="primary200">
+              <Clock width={pxToRem(40)} height={pxToRem(40)} />
+            </IconBox>
+          )}
+          <Box marginTop={2}>
+            <Link href="https://docs.strapi.io" isExternal>
+              {formatMessage({
+                id: 'global.documentation',
+                defaultMessage: 'Read the documentation',
+              })}
+            </Link>
+          </Box>
+        </Overlay>,
+        // eslint-disable-next-line no-undef
+        globalThis?.document?.body
```

Comment:
If this is executed you know all of these properties exist.

```suggestion
        globalThis.document.body
```

Reply:
Sure, if we move to SSR we'll have to revert but idm 😄